pallets/ddc-verification/src/lib.rs

```diff
@@ -2892,40 +2917,65 @@ pub mod pallet {
 		/// - `cluster_id`: cluster id of a cluster
 		/// - `dac_nodes`: List of DAC nodes
 		pub(crate) fn get_era_for_validation(
-			// todo! this needs to be rewriten - too complex and inefficient
 			cluster_id: &ClusterId,
 			dac_nodes: &[(NodePubKey, StorageNodeParams)],
 		) -> Result<Option<EraActivity>, OCWError> {
-			let current_validator_data = Self::fetch_current_validator()?;
+			let this_validator_data = Self::fetch_current_validator()?;
+			let this_validator = T::AccountId::decode(&mut &this_validator_data[..]).unwrap();
```

Comment:
Let's use `expect` instead of `unwrap` here and in the other updated places?

Reply:
Done